pom.xml

```diff
@@ -1258,6 +1264,8 @@
                 launched by the tests have access to the correct test-time classpath.
               -->
               <SPARK_DIST_CLASSPATH>${test_classpath}</SPARK_DIST_CLASSPATH>
+              <SPARK_PREPEND_CLASSES>1</SPARK_PREPEND_CLASSES>
```

Comment:
yes, that's the gist of it. Some code needed to be added (especially in YarnClusterSuite) to also propagate that to child processes, but this is the main idea.